### PR TITLE
Improved Travis instructions

### PR DESCRIPTION
**Pull request by snoyberg**
* Consolidate into a single page (with reference from guide)
* Put content into external files for easier copy/paste
* Include a simple configuration
* Give advice on how to choose the right configuration

### CODE REVIEW

File: doc/GUIDE.md [hunk 1]
```diff
@@ -1939,213 +1939,8 @@ can use the `stack dot` command and Graphviz. More information is
 
 ### Travis with caching
 
-Many people use Travis CI to test out a project for every Git push. We have [a
```
Comment (Blaisorblade) [inline doc/GUIDE.md]: There are still references to this section, e.g. in https://github.com/commercialhaskell/stack/pull/2578/files#diff-e96fbd8ff8894442360d3de9bdd33895R86, and at least that is now stale even with the redirection.

File: doc/travis_ci.md [hunk 2]
```diff
@@ -1,11 +1,32 @@
...
+* [The complex Travis configuration](https://raw.githubusercontent.com/commercialhaskell/stack/master/doc/travis-complex.yml)
+  is intended for projects that need to support multiple GHC versions
+  and multiple OSes, such as open source libraries to be released to
+  Hackage. It tests against cabal-install, as well as Stack on Linux
+  and OS X. The configuration is significantly more involved to allow
+  for all of this branching behavior.
+
+Each of these configurations is ready to be used immediately, just
+copy-paste the content into the `.travis.yml` file in the root or your
+repo, enable Travis on the repo, and you're good to go.
```
Comment (Blaisorblade) [inline doc/travis_ci.md]: I'd point out that the latter configuration might often need customization. It makes choices on which GHC versions to support, and assumes that the same `stack.yaml` (including extra-deps) works on all the chosen resolvers—that's likely if no extra-deps are used. Maybe add a pointer to the relevant sections below?

File: doc/GUIDE.md (hunk 1 above)
Comment (snoyberg) [inline doc/GUIDE.md]: Good point, I'll do a grep to find all such places in the repo.

**Comment by Blaisorblade** [comment]
Looks like a big improvement! BTW there's progress on the Travis side, though they're not yet done it seems:
https://github.com/travis-ci/apt-package-whitelist/issues/379#issuecomment-246109122

File: doc/GUIDE.md (hunk 1 above)
Comment (snoyberg) [inline doc/GUIDE.md]: Commit pushed. I couldn't follow the link you provided, but I think I found the only two references in the repo.

**Comment by snoyberg** [comment]
Yeah, I saw some of the Travis activity, but honestly gave up following the back-and-forth at some point. Glad it's moving forward.

And yes, I think this is a big improvement. Beyond including the simple config (which was definitely a good idea on your part), this scratches an itch of mine by putting the configs in standalone files for easier addition to project by just `curl`ing.

File: doc/travis_ci.md (hunk 2 above)
Comment (snoyberg) [inline doc/travis_ci.md]: Good call, added a paragraph.

**Comment by Blaisorblade** [comment]
Oh, the Travis failure seems just "master stopped building pre-AMP". On it.

But the PR itself LGTM now.

**Pull request merged**